File: symfony/src/Entity/Players/CraftingIngredient.php
<?php

namespace App\Entity\Players;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use App\Repository\Players\CraftingIngredientRepository;
use App\Entity\Cards\Card;

#[ORM\Entity(repositoryClass: CraftingIngredientRepository::class)]
#[ORM\Table(name: 'crafting_ingredient')]
class CraftingIngredient
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups(['craftingIngredient:read'])]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Groups(['craftingIngredient:read', 'craftingIngredient:write'])]
    private int $quantity = 1;

    #[ORM\ManyToOne(targetEntity: CraftingRecipe::class, inversedBy: 'ingredients')]
    #[ORM\JoinColumn(nullable: false)]
    private ?CraftingRecipe $recipe = null;

    #[ORM\ManyToOne(targetEntity: Card::class, inversedBy: 'used_in_recipes')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Card $card = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getQuantity(): int
    {
        return $this->quantity;
    }

    public function setQuantity(int $quantity): static
    {
        $this->quantity = $quantity;
        return $this;
    }

    #[Groups(['craftingIngredient:read'])]
    public function getRecipeId(): ?int
    {
        return $this->recipe?->getId();
    }

    public function getRecipe(): ?CraftingRecipe
    {
        return $this->recipe;
    }

    public function setRecipe(?CraftingRecipe $recipe): static
    {
        $this->recipe = $recipe;
        return $this;
    }

    #[Groups(['craftingIngredient:read'])]
    public function getCardId(): ?int
    {
        return $this->card?->getId();
    }

    public function getCard(): ?Card
    {
        return $this->card;
    }

    public function setCard(?Card $card): static
    {
        $this->card = $card;
        return $this;
    }

}

File: spring/src/main/java/cardsproject/domain/players/CraftingIngredient.java
package cardsproject.domain.players;

import jakarta.persistence.*;

@Entity
@Table(name = "crafting_ingredients")
public class CraftingIngredient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer quantity = 1;

    // @ManyToOne -> CraftingRecipe, onDelete=CASCADE, relatedName=ingredients
    @Column(name = "recipe_id")
    private Long recipeId;
    // @ManyToOne -> Card, onDelete=PROTECT, relatedName=used_in_recipes, via=cards
    @Column(name = "card_id")
    private Long cardId;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Integer getQuantity() { return quantity; }
    public void setQuantity(Integer quantity) { this.quantity = quantity; }
    public Long getRecipeId() { return recipeId; }
    public void setRecipeId(Long recipeId) { this.recipeId = recipeId; }
    public Long getCardId() { return cardId; }
    public void setCardId(Long cardId) { this.cardId = cardId; }
}
